fud2/src/lib.rs

```diff
@@ -873,12 +868,23 @@ pub fn build_driver(bld: &mut DriverBuilder) {
         let data_path = e.external_path(data_name.as_ref());
         e.var("sim_data", data_path.as_str())?;
 
-        // Cocotb is wants files relative to the location of the makefile.
+        // Cocotb wants files relative to the location of the makefile.
         // This is annoying to calculate on the fly, so we just copy necessary files to the build directory
         e.rule("copy", "cp $in $out")?;
-        e.rule("make-cocotb", "make DATA_PATH=$sim_data VERILOG_SOURCE=$in COCOTB_LOG_LEVEL=CRITICAL > $out")?;
-        // This cleans up the extra `make` cruft, leaving what is in between `{` and `}.`
-        e.rule("cleanup-cocotb", r"sed -n '/Output:/,/make\[1\]/{/Output:/d;/make\[1\]/d;p}' $in > $out")?;
+
+        let waves = e.config_constrained_or("waves", vec!["true", "false"], "false")?;
+        let waves = FromStr::from_str(&waves).expect("The 'waves' flag should be either 'true' or 'false'.");
+        if waves{
+            //adds lines based on what is needed for icarus fst output.
+            e.rule("iverilog-fst-sed",
```

Comment:
Wow, pretty wild. Using sed to splice in the appropriate code to dump out the trace is definitely impressive!!

There may another way: simulators like Icarus can take command-line arguments. This is how we toggle tracing in the "normal" test workflow, for example:
https://github.com/calyxir/calyx/blob/a4c2442675d3419be6d2f5cf912aa3f804b3c4ab/fud2/rsrc/tb.sv#L48-L55

I don't know if Cocotb provides a way to pass CLI arguments to the simulation, however. Maybe this would require hacking the relevant Makefile?

Reply:
Yeah I think this `sed` approach is a bit unfortunate as it looks pretty gnarly and is hard to really "get", for me at least. 

Want to make sure I'm understanding correctly, passing in command-line arguments would move the `sed` insertion to one of the `verilog` backends, so that when we do [axi-wrapped-]calyx -> verilog the appropriate `$dump*` system functions are present, and simply toggled based on said argument? If that's the case I think leaving this as a `sed` insertion is just as good as changing the backends. This could probably also be done as a simple concatenation as well, as the location of the `$dump*` system functions doesn't matter from what I can tell. So a simple `>>` might be the best way forward.

Reply:
Spoke with @sampsyo synchronously, going to leave this as is for now. In the future it could be nice to let `cocotb` handle this insertion stuff for us? I vaguely recall this being possible by simply passing in [`WAVES=1`](https://docs.cocotb.org/en/stable/simulator_support.html#waveforms) during the `make` invocation, but looked around and couldn't find much about `cocotb` automatically adding the required system functions.